modules/tensor_mechanics/src/materials/ComputeTemperatureDependentIsotropicElasticityTensor.C

```diff
@@ -0,0 +1,53 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ComputeTemperatureDependentIsotropicElasticityTensor.h"
+
+#include "Function.h"
+
+template<>
+InputParameters validParams<ComputeTemperatureDependentIsotropicElasticityTensor>()
+{
+  InputParameters params = validParams<ComputeElasticityTensorBase>();
+  params.addClassDescription("Compute an isotropic elasticity tensor for elastic constants that change as a function of temperature");
+  params.addRequiredCoupledVar("temperature", "Coupled temperature");
+  params.addRequiredParam<FunctionName>("youngs_modulus_function","Function defining Young's modulus as a function of temperature");
+  params.addRequiredParam<FunctionName>("poissons_ratio_function","Function defining Poisson's ratio as a function of temperature");
```

Comment:
Worlds collide here. In tensor mechanics (or phase field) we do not use functions in that way anywhere. I would have implemented this as a material property (and allowed for arbitrary variable dependencies). There is nothing that is specific to "Temperature" in this class at all.

The way this class is implemented users might be tempted to abuse it (in absence of a `ComputeVariableIsotropicElasticityTensor` class). The problem is that this class does not provide derivatives of the elasticity tensor, which has the potential to break stuff.